### PR TITLE
Factory Initializer Objects

### PR DESCRIPTION
**Pull request by pbauman**
Ala libqueso/queso#508, this creates FactoryInitializer objects that statically instantiate the factory classes so they don't get stripped by static linking. I collected all of them into `SimulationInitializer`, which needs to be instantiated before `Simulation`. (A forthcoming PR is going to refactor the GRINS `main` to just call a function which does all the setup and run to ease extension by users.) I believe this captures all "properly" factoried objects, which is confirmed by building with `--enable-static --disable-shared` at configure time (I also confirmed we actually failed in this case before this PR).  We should add a CIVET recipe for this build configuration (#469, #472).

### CODE REVIEW

**Comment by pbauman** [comment]
Will merge this afternoon barring any complaints.

File: test/exact_soln/vorticity_qoi.C [hunk 1]
```diff
@@ -70,7 +71,9 @@ int main(int argc, char* argv[])
 
   // Initialize libMesh library.
   libMesh::LibMeshInit libmesh_init(argc, argv);
- 
+
+  GRINS::SimulationInitializer initializer;
+
   GRINS::SimulationBuilder sim_builder;
```
Comment (roystgnr) [inline test/exact_soln/vorticity_qoi.C]: Could we have SimulationBuilder create a SimulationInitializer?  Then we wouldn't need to change so much user code.

Comment (pbauman) [inline test/exact_soln/vorticity_qoi.C]: My intent is for SimulationBuilder to go away after properly factorying out Solver, QoI, etc. (goal is to finish that before tomorrow morning). Also, the PR I'm working on right now will hide all this stuff from the user.

**Comment by pbauman** [comment]
Runner PR (#486) hides this away, so I'm going ahead and merging.

**Pull request merged**